Sources/SafeDICore/Models/TypeDescription.swift

```diff
@@ -27,7 +27,7 @@ public enum TypeDescription: Codable, Hashable, Comparable, Sendable {
     /// A nested type with possible generics. e.g. Array.Element or Swift.Array<Element>
     indirect case nested(name: String, parentType: TypeDescription, generics: [TypeDescription])
     /// A composed type. e.g. Identifiable & Equatable
-    indirect case composition([TypeDescription])
+    indirect case composition(Set<TypeDescription>)
```

Comment:
A downside of using a `Set` here is that we no longer know the written order of the types of the composition in the source code. We _could_ offset that by utilizing an array still, and then doing set comparisons in a manually written `==` implementation, but:

1. Doing just-in-time `Set` comparisons can get expensive when doing a lot of comparisons
2. Maintaining a manually written `==` implementation sounds... not fun

So I'm pretty bullish that this approach is good enough for our needs. Open to feedback and alternative thoughts of course.

Reply:
There's always `OrderedSet` from `swift-collections`, but then you'd be adding a dependency.

Reply:
yup! `OrderedSet` would solve the just-in-time `Set` comparison problem. But I'd still need the manually written `==` which is a bit painful. Might be worth it though?

Reply:
I have no qualms with a dependency, especially one that already supports Linux 😄 https://swiftpackageindex.com/apple/swift-collections

Reply:
I'm gonna write a new type that houses both an array and a set, with its own `==` implementation, and use that.